db/database_test.go

```diff
@@ -1850,32 +1850,38 @@ func TestChannelQuery(t *testing.T) {
 	// Create a doc to test removal handling.  Needs three revisions so that the removal rev (2) isn't
 	// the current revision
 	removedDocID := "removed_doc"
-	removedDocRev1, _, err := collection.Put(ctx, removedDocID, body)
+	removedDocRev1, rev1, err := collection.Put(ctx, removedDocID, body)
 	require.NoError(t, err, "Couldn't create removed_doc")
-	removalSource, removalVersion := collection.GetDocumentCurrentVersion(t, removedDocID)
 
 	updatedChannelBody := Body{"_rev": removedDocRev1, "key1": "value1", "key2": 1234, "channels": "DEF"}
-	removalRev, _, err := collection.Put(ctx, removedDocID, updatedChannelBody)
+	removalRev, rev2, err := collection.Put(ctx, removedDocID, updatedChannelBody)
 	require.NoError(t, err, "Couldn't update removed_doc")
 
 	updatedChannelBody = Body{"_rev": removalRev, "key1": "value1", "key2": 2345, "channels": "DEF"}
-	removedDocRev3, _, err := collection.Put(ctx, removedDocID, updatedChannelBody)
+	_, rev3, err := collection.Put(ctx, removedDocID, updatedChannelBody)
 	require.NoError(t, err, "Couldn't update removed_doc")
 
+	log.Printf("versions: [%v %v %v]", rev1.HLV.Version, rev2.HLV.Version, rev3.HLV.Version)
+
+	// TODO: check the case where the channel is removed with a putExistingRev mutation
```

Comment:
Is this TODO for this PR or a follow-up? I guess it might be waiting for the fixes to the conflict detection functions?

Reply:
Yes - I wasn't able to fully test this yet, will make a note to fix this in the push replication PR.